### PR TITLE
Tags Tables

### LINKED ISSUE #928: Add additional database tables for tags
Based on the [tags design](https://github.com/NASA-AMMOS/aerie-ui/issues/604), we will probably need to add one or more additional tag tables to store global tags.

### PR DESCRIPTION
**Pull request by Mythicaeda**
* **Tickets addressed:** Closes #928
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

- `tags` table was created in new schema, `metadata`
- To enforce FK constraints, association tables were added and extracted from existing `tags text[]` columns
- Plan Merge components were reorganized to match how the rest of the DB is set up, with long functions being pulled into their own files
- Schema declaration was pulled out into a `schemas.sql` file in order to avoid order-sensitivity between files creating objects in the same non-public schema (ie the tags association tables and hasura functions)
- the `tags` table introduces two reserved owners: `Mission Model` and `Aerie Legacy`. `Mission Model` are tags that will come from #934. `Aerie Legacy` are tags that existed in the Merlin DB prior to the up migration in this PR being run

## Verification

- Shared Merlin code in the DBTests was pulled into a helper class, `MerlinDatabaseTestHelper`. I considered "shared" to mean "used in testing multiple features".
- Existing PlanMerge tests were updated to reflects tags no longer being a field on activity directives or snapshot activities
- DB Tests were added for Tags

## Documentation

As far as I can find, Tags are not included in the existing docs, so no documentation changes need to be made.

### CODE REVIEW

File: deployment/hasura/migrations/AerieSequencing/5_tags/up.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+create schema metadata;
+create table metadata.expansion_rule_tags (
+  rule_id integer references public.expansion_rule
+    on update cascade
+    on delete cascade,
+  tag_id integer not null unique,
```
Comment (Mythicaeda) [inline deployment/hasura/migrations/AerieSequencing/5_tags/up.sql]: remove `unique`

File: deployment/hasura/migrations/AerieScheduler/8_tags/up.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+create schema metadata;
+create table metadata.scheduling_goal_tags (
+  goal_id integer references public.scheduling_goal
+    on update cascade
+    on delete cascade,
+  tag_id integer not null unique,
```
Comment (Mythicaeda) [inline deployment/hasura/migrations/AerieScheduler/8_tags/up.sql]: remove `unique`

File: deployment/hasura/metadata/databases/AerieMerlin/tables/metadata/activity_directive_tags.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - role: viewer
+    permission:
+      columns: [plan_id, directive_id, tag_id]
+      filter: {}
+      allow_aggregations: true
+insert_permissions:
+  - role: user
+    permission:
+      columns: [plan_id, directive_id, tag_id]
+      check: {}
```
Comment (Mythicaeda) [inline deployment/hasura/metadata/databases/AerieMerlin/tables/metadata/activity_directive_tags.yaml]: restrict to plan owner/collaborator

File: deployment/hasura/metadata/databases/AerieMerlin/tables/metadata/plan_tags.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  - role: viewer
+    permission:
+      columns: [plan_id, tag_id]
+      filter: {}
+      allow_aggregations: true
+insert_permissions:
+  - role: user
+    permission:
+      columns: [plan_id, tag_id]
+      check: {}
```
Comment (Mythicaeda) [inline deployment/hasura/metadata/databases/AerieMerlin/tables/metadata/plan_tags.yaml]: make owner

**Review by camargo — APPROVED** [review]
Looks and works great! Thanks for spending the time and writing such detailed tests. Good to go.

**Pull request merged**